### DISCUSSION TITLE
Articulating What Data and Information Is and Is Not in Scope for NODE

### DISCUSSION THREAD

**Discussion by kswb** [NODE-Collective/.github#3] [Ideas]
## Background

MOVED from spec discussion to org discussion. Last week at the NODE data subcommittee meeting we agreed that it would be helpful for the community writ large if NODE articulates what is and is not in scope with respect to incentives information. Doing so would create clarity and help focus discussions, issues, and other community requests more productively.

## First Attempt at Content

The first idea is that we could create a section of the specification document, like an FAQs section, a separate document like a readme, a dedicated FAQs page, pinned blog post on NODE, or anywhere else that has high visibility.

Below is a first attempt that pulls from conversations between the Pencil Energy team and the NODE data working group.

| Concept, Data, or Other Item | In-Scope? | Description |
| -- | -- | --|
|NODE-built software tools for data processing, editing, and maintenance | No | A GUI for editing and outputting data in a standard format is not in scope.|
|Calculating expected savings from incentives | No | Calculations or savings scenarios are not supported in this iteration of NODE [and may be a future roadmap item?].|
|Determining customer eligibility for incentive programs | No | Eligibility requirements such as hardware specifications or income levels will be presented in free text. [Structured requirements may be a future roadmap item?].|
|Stackability definitions | No | Creating a standardized methodology for stacking or layering programs is out of scope for NODE [and may be a future roadmap item?].|
|Finding relevant programs for me or my customers | No | NODE will not identify the applicable program area currently and as of now does not have plans to build tools for this.|
|Completing paperwork and submitting applications | No | NODE will support paperwork or application management and as of now does not have plans to build tools for this.|

### Specific Requests For Feedback in This Discussion Thread

* Whether this discussion is correctly identified as "Ideas"
* The ideas and general goal of this post
* The first cut of content included in the below table
* Consensus on next steps on whether this information should be included and, if so, where

Thanks!

Kyle

**Comment by oyamauchi** [comment]
Going to do these out of order:

* **Software tools**: In general, I think tools that operate on NODE-format data can be in scope. I've actually prototyped a GUI editor for NODE data, having realized that it's not straightforward to generate files in this format without either a program specifically meant for it, or a relational database. Which reminds me, now that we have a Github org I should put the code on here.

  Even if we don't end up turning that prototype into a real supported product, I do think there would be value in releasing a dataset validator. The intended user would be a data maintainer, or maybe a program administrator who wants to publish their program's incentives in NODE format. We could release it as a simple stateless web service, as well as in source code form.

   We may also end up building a web service to host the data, with an API to serve it. The question of where NODE data is going to live is still unresolved, so I don't know if we'll do this.

* **Stackability**: I'd say "not now, but probably soon". RA's internal data already has this (we have a concept of relationships between incentives).
  
  I have to admit I've never quite understood what's difficult about this — it seems to me that the baseline assumption should be that any two incentives can be claimed together ("stacked"), and counterexamples are actually quite rare? — but I'm ready to be proven wrong. Also, people keep saying the phrase "stacking and braiding" and I don't understand what the distinction between those two things is.

The other four items have unearthed an assumption I had in my brain all along, which is that NODE's purview is limited to the dataset — its schema and its contents — and end-uses of the data are out of scope.

However, the schema is certainly _influenced_ by expected end-uses. (That's actually a problem we've had: some schema-design decisions have been hard to make without a specific end-use to design for.) So I would say the answer to "would NODE evolve the schema to support a data consumer doing use case X?" varies depending on the value of X, whereas the answer to "would NODE do use case X itself?" is no for any X that isn't maintaining the schema and dataset.

Here's what I think that means for the other four items you brought up:

* **Determining eligibility**: I think we will probably want to evolve the schema to support this at least to some extent. RA already has detailed income-qualification information, which is unfortunately pretty tricky to represent in a relational format like NODE's. (E.g. [this kind of thing](https://www.efficiencyvermont.com/income-limit). Although we store these in a relational DB, we "cheat" by putting the meat of the info on income thresholds into a column of [type JSONB](https://www.postgresql.org/docs/current/datatype-json.html).)

  I can imagine adding a way to represent equipment-spec restrictions (e.g. "HSPF2 >= 8.0") as well, although we at RA have not tackled that yet.

* **Calculating savings**: I don't know off the top of my head how much more detailed the data would need to be (beyond what the "determining eligibility" item would require) to support this. I think we should be open to supporting this use case, since it's a very obvious one.

* **Finding relevant programs**: this might be right on the edge of the scope definition. If we host NODE data first-party (as I alluded to in the "software tools" part) then I think this could be in scope, as it could be considered as a way to query the dataset. Otherwise, probably not.

  (Note: I think I may not have fully understood what you mean with this point.)

* **Completing paperwork**: definitely something we should expect a data consumer to build on top of the NODE dataset. The answer to whether we'd evolve the schema to support this depends on how detailed the data would need to be, I think. It seems to me that if it were detailed enough to support the "calculating savings" use case, then it could probably support this, but I'm not sure. I'd be interested to hear from ELI (@FredericoAndrade) about that, since I believe they're actually doing this.

**Comment by kswb** [comment]
In order:
- I'm excited to see the prototype for the GUI! Great idea on the dataset validator. Will NODE members/volunteers serve as data maintainers? Would that be a NODE internal task? Also looking forward to learning where the data will live.
- On stackability, yes, Stackability isn't complex for most cases, but there are a quite a few counter examples we've come across so I wouldn't call it rare. The issue is that the counterexamples are so complex that it is a massive headache to deal with. In California you can't combine programs that draw from the same public program charge funding source which is an added charge on utility bills. Those funds get spread to so many orgs and their programs without much visibility, so today it requires reading T&C documents to find out whether they have that funding source. Stacking and braiding are identical, but I like to say "stack". Comes from my MTG and gaming days 😄 

On thinking about end-uses:
- Looking forward to contributing on this one since this was one of our primary focuses with our schema and approach to the API design! Eligibility is interesting and I think we could meet in the middle relative to our solution, pulling only the most critical elements like income, equipment, contractor network membership, etc.
- We can discuss calculating savings at the same time. What created the most complexity for us was realizing that certain eligibility "pathways" come with certain dollar savings values that are mutually exclusive with others. That's where a lot of the complexity can exist.
- This point refers to discoverability for a given customer, so that could be a location or address search like a generated list. This could be duplicative with RA's calculator, but the address-level precision is something that could be added.
- Makes sense regarding paperwork!
